fs/littlefs/syscfg.yml

```diff
@@ -0,0 +1,50 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+syscfg.defs:
+    LITTLEFS_FLASH_AREA:
+        description: 'Flash area to use for littlefs.'
+        type: flash_owner
+        value:
+        restrictions:
+            - $notnull
+
+    LITTLEFS_DETECT_FAIL:
+        description: 'Controls behaviour when encountering corrupt littlefs area.'
+        value: 'LITTLEFS_DETECT_FAIL_FORMAT'
```

Comment:
I do not like using preprocessor definitions as syscfg values as effectively this injects raw source code via syscfg setting. This should better be 'choice' insead.

Reply:
Done.